src/controllers/posts.ts

```diff
@@ -0,0 +1,87 @@
+import { Request, Response } from 'express';
+import axios from 'axios';
+import { ServerPostData } from '../interfaces/serverPostData';
+import { ServerPostArray } from '../interfaces/serverPostArray';
+import { ServerUserArray } from '../interfaces/serverUserArray';
+import { ServerUserElement } from '../interfaces/serverUserElement';
+import { User } from '../models/users';
+import { Posts } from '../models/posts';
+
+
+export class PostController {
+    domain:string = "https://jsonplaceholder.typicode.com/";
+    res:Response;
+    req:Request;
+
+    constructor (res:Response, req: Request) {
+        this.res = res;
+        this.req = req;
+    }
+
+convertPosts(paramUser: ServerUserElement,paramPostArray: ServerPostArray):Array<any> {
+    //console.log(paramUserArray.address)
+    let responseArray = paramPostArray.data.map((element)=>{
+        const nameSeparator = paramUser.data.name.split(" ");
+        let prefix:string = "";
+        let firstName:string = "";
+        let lastName:string = "";
+
+        if (nameSeparator.length > 2) {
+            prefix = nameSeparator[0];
+            firstName = nameSeparator[1];
+            lastName = nameSeparator[2];
+        }
+        else
+        {
+            prefix = "";
+            firstName = nameSeparator[0];
+            lastName = nameSeparator[1];
+        }
+     
+    let postMap:Posts = {
+        userId:element.userId,
+        name: firstName +" "+ lastName,
+        email:  paramUser.data.email,
+        postId: element.id,
+        title: element.title,
+        body: element.body
+    };
+    //userMap.id = element.id
+    return postMap;
+});
+return responseArray
+}
+
+    async getUserPosts():Promise<boolean> {    
+        let userId:string = this.req.params.userId;
+        const userUrl:string = this.domain + "users/" + userId;
```

Comment:
```
const userUrl: string = `${this.domain}users/${userId}`
```
User string interpolation instead of `+`

Reply:
done